content/flux/v0.x/stdlib/universe/difference.md

```diff
@@ -24,17 +24,19 @@ _**Output data type:** Float_
 
 ```js
 difference(
-  nonNegative: false,
-  columns: ["_value"],
-  keepFirst: false
+    nonNegative: false,
+    columns: ["_value"],
+    keepFirst: false,
+    initialZero: false,
 )
 ```
 
 ## Parameters
 
 ### nonNegative {data-type="bool"}
 Indicates if the difference is allowed to be negative.
-When set to `true`, if a value is less than the previous value, it is assumed the previous value should have been a zero.
+When set to `true`, if a value is less than the previous value, the function
```

Comment:
_When set to `true`, if a value is less than the previous value, the function assumes the previous value should have been a zero._

to be sure, `nonNegatives = true` means diff is allowed to be negative? 

Curious why if the first value is 3 and second value is 2, why would the function assume 3 should be zero?

Reply:
> to be sure, `nonNegatives = true` means diff is allowed to be negative?

No, when set to true, it returns a zero-sum change from the previous. This is a special application specifically designed so `increase()` correctly supports counter data.